### DISCUSSION TITLE
mac vs net{idx}/mac

### DISCUSSION THREAD

**Discussion by r00ta** [ipxe/ipxe#995] [Q&A] [answered]
Hi there!

I have a set of machines with multiple interfaces (`net0`, `net1`, `net2`, `net3`) and they pxe boot from one of them (some will boot from `net0`, some from `net1` and so on). My DHCP server sends then an ipxe config file and I need the machines to communicate back the mac address of the interface used for dhcp. 

I ended up with the following script

```
set interface_number:int32 0
:find_interface_with_address 
    isset ${net${interface_number}/ip} || goto find_interface_with_address_done
    inc interface_number && goto find_interface_with_address
:find_interface_with_address_done
chain http://${next-server}:5248/ipxe.cfg-${net${interface_number}/mac}
```

But then I found that the setting `mac` seems to always point to the mac address of the interface used for DHCP. So I could simply write

```
chain http://${next-server}:5248/ipxe.cfg-${mac}
```

Could anybody confirm that my understanding is correct and that `mac` always contain the MAC address used for DHCP? 

Thank you very much :)

**Comment by NiKiZe** [comment, marked as answer]
The top level settings inherits from the subnodes.
It is not always the one from DHCP, but rather the "last used"

But what you want here is netX/mac which is the last successful opened interface.
And I do mean netX, you should not replace the X with a number.

If you only want one nic, you might want to use snponly instead, which will always only have the nic that iPXE was chained thru.

**Reply by NiKiZe** [reply]
The netX documentation is maybe a bit lacking, but the hierarchy is documented at https://ipxe.org/settings

**Reply by r00ta** [reply]
Thank you @NiKiZe for the super fast reply! Is there a command to use that `snponly` utility to extract the NIC that was used for the pxe boot? Sorry for the dumb question, I'm new to ipxe :)

**Reply by r00ta** [reply]
just to double check my understanding: also `next-server` comes from the latest used interface right?

**Reply by NiKiZe** [reply]
by snponly, I mean that you use snponly.efi (for example the one from http://boot.ipxe.org/snponly.efi) instead of ipxe.efi which I assumed you are using currently, if you are booting in pcbios mode then the similar variant is undionly.kpxe.

To avoid confusion, always use the full scoped variable, that is netX/next-server

**Reply by r00ta** [reply]
Thank you very much @NiKiZe !

**Comment by bdrung** [comment]
Is `netX` documented anywhere? Yesterday I searched for an equivalent to "ipappend 2" from syslinux, but I failed to find ipxe relevant documentation.

**Reply by NiKiZe** [reply]
I do apologize for initially reading this comment as being from OP (small screen on phone made me miss it)
After searching I do agree that netX documentation seems a bit lacking, It should probably be covered by the documentation as mentioned in https://github.com/ipxe/ipxe/discussions/995#discussioncomment-6425978 Other than that forum posts etc do document it's use.

And yes the concept of "use variables in your script" is a bit different from the more limiting syslinux syntax, iPXE aims at having simple building blocks that can be used in more complex scenarios.
